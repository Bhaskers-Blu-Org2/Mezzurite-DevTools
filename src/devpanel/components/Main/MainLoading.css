@media only screen {
  .capture-cycle--skeleton {
    padding-bottom: 4px;
  }

  .capture-cycle--skeleton-component {
    border-radius: 8px;
    height: 28px;
    margin: 0 16px 16px;
  }

  .capture-cycle--skeleton-title {
    height: 24px;
    border-radius: 8px;
    margin: 16px auto 36px;
    max-width: 50%;
  }

  .shimmer {
    background: var(--microsoft-gray-lightest);
    background-image: -webkit-gradient(linear, left top, right top, from(var(--microsoft-gray-lightest)), color-stop(20%, var(--microsoft-gray-lighter)), color-stop(60%, var(--microsoft-gray-lightest)), to(var(--microsoft-gray-lightest)));
    background-image: -webkit-linear-gradient(left, var(--microsoft-gray-lightest) 0%, var(--microsoft-gray-lighter) 20%, var(--microsoft-gray-lightest) 60%, var(--microsoft-gray-lightest) 100%);
    background-image: -o-linear-gradient(left, var(--microsoft-gray-lightest) 0%, var(--microsoft-gray-lighter) 20%, var(--microsoft-gray-lightest) 60%, var(--microsoft-gray-lightest) 100%);
    background-image: linear-gradient(to right, var(--microsoft-gray-lightest) 0%, var(--microsoft-gray-lighter) 20%, var(--microsoft-gray-lightest) 60%, var(--microsoft-gray-lightest) 100%);
    background-repeat: no-repeat;
    background-size: 200vw 200vw;
    opacity: 0.6;
    -webkit-animation-duration: 2s;
    -webkit-animation-fill-mode: forwards;
    -webkit-animation-iteration-count: infinite;
    -webkit-animation-name: placeholderShimmer;
    -webkit-animation-timing-function: linear;
  }
}

@media only screen and (min-width: 540px) {
  .capture-cycle--skeleton-title {
    margin: 16px 16px 36px;
    max-width: 30%;
  }
}

@-webkit-keyframes placeholderShimmer {
  0% {
    background-position: -100vw 0;
  }

  100% {
    background-position: 100vw 0;
  }
}
