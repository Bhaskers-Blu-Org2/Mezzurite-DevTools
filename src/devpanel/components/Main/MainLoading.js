import React from 'react';

import './MainLoading.css';

const MainLoading = () => (
  <div className='capture-cycle--card capture-cycle--skeleton'>
    <div className='capture-cycle--skeleton-title shimmer' />
    <div className='capture-cycle--skeleton-container'>
      <div className='capture-cycle--skeleton-component shimmer' />
      <div className='capture-cycle--skeleton-component shimmer' />
      <div className='capture-cycle--skeleton-component shimmer' />
    </div>
  </div>
);

export default MainLoading;
