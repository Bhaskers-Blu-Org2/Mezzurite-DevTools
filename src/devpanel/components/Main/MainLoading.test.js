import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MainLoading from './MainLoading';

describe('MainLoading.js', () => {
  const renderer = new ShallowRenderer();

  it('should render the main loading view', () => {
    const tree = renderer
      .render(<MainLoading />);
    expect(tree).toMatchSnapshot();
  });
});
