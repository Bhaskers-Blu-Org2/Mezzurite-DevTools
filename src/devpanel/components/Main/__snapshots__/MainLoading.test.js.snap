// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLoading.js should render the main loading view 1`] = `
<div
  className="capture-cycle--card capture-cycle--skeleton"
>
  <div
    className="capture-cycle--skeleton-title shimmer"
  />
  <div
    className="capture-cycle--skeleton-container"
  >
    <div
      className="capture-cycle--skeleton-component shimmer"
    />
    <div
      className="capture-cycle--skeleton-component shimmer"
    />
    <div
      className="capture-cycle--skeleton-component shimmer"
    />
  </div>
</div>
`;
